### ISSUE TITLE
[Visual/Info] ScriptNode Parameter Value Not Updating When Triggered From the Interface Preview

### ISSUE THREAD

**Issue by aaronventure** [christophhart/HISE#572]
<img width="248" alt="image" src="https://github.com/user-attachments/assets/7be210ca-b7fb-49a0-9b41-7471946f3902">

![scriptnodeparameterlabel](https://github.com/user-attachments/assets/4b79ca25-6732-46cb-9051-0d9b8c57b123)

macOS 14.5
cd51dc7bc8d5f539184a4308b41c6b5cabc2aaaa

**Comment by christoph-hart** [comment]
Is this still happening? I've refactored quite a bit in the scriptnode workspace and I can't reproduce it here.

**Comment by christoph-hart** [comment]
Hmm, that's surprisingly difficult to solve.

The issue is that calling setAttribute() is in fact not updating the value tree property of the network but directly forward the call to whatever target the knob is set (and the knob ring fetches the value from there, hence it's differenc than the text property (also when you start dragging, the slider jumps to another value which I also experienced quite often but couldn't reliably reproduce).

The "fix" would be pretty trivial: make setAttribute() update the value tree property (all slider callbacks & UI updates are listening to property changes of the value tree. However this procedure is more heavyweight than it is right now (it involves undo support & string lookups to find the property to overwrite so if you use that in a realtime context (eg. in a macro modulation chain) you'll get worse performance than it is currently the case (I think that's why I implemented it like this in the first place).

I'll see if I can keep the way it works internally but hack around so the UI will update correctly when you hover.
